crates/http-sink/src/sink.rs

```diff
@@ -28,38 +31,56 @@ impl HttpSink {
             request = request.header(key, value.trim());
         }
 
-        Ok(Self { request })
-    }
-}
-
-#[async_trait]
-impl Sink<String> for HttpSink {
-    async fn connect(self, _offset: Option<Offset>) -> Result<LocalBoxSink<String>> {
-        let request = self.request;
-        let unfold = futures::sink::unfold(
+        Ok(Self {
+            client,
             request,
-            |mut request: RequestBuilder, record: String| async move {
-                tracing::trace!("{:?}", request);
-
-                request = request.body(record);
-                let response = request
-                    .try_clone()
-                    .ok_or(anyhow!("ERR: Cannot clone request"))?
-                    .send()
-                    .await?;
+            url_parameters: config.url_parameters.clone(),
+        })
+    }
 
-                if response.status().is_success() {
-                    tracing::debug!("Response Status: {}", response.status());
-                } else {
-                    tracing::warn!("Response Status: {}", response.status());
-                    tracing::debug!("{:?}", response);
+    pub(crate) fn make_request(&self, record: &ConsumerRecord) -> Result<RequestBuilder> {
+        let mut builder = self
+            .request
+            .try_clone()
+            .ok_or(anyhow!("ERR: Cannot clone request"))?;
+
+        if !self.url_parameters.is_empty() {
+            let str = String::from_utf8(record.as_ref().to_vec())?;
+            if let Ok(json_message) =
+                serde_json::from_str::<HashMap<String, serde_json::Value>>(&str)
+            {
+                for param in self.url_parameters.iter() {
+                    let url_key = param.url_key.clone().unwrap_or(param.record_key.clone());
+                    if json_message.contains_key(&param.record_key) {
+                        let mut value = json_message.get(&param.record_key).unwrap().to_string();
+                        if let Some(ref prefix) = param.prefix {
+                            value = prefix.clone() + &value;
+                        }
+                        if let Some(ref suffix) = param.suffix {
+                            value = value.clone() + &suffix;
+                        }
+                        builder = builder.query(&[(encode(&url_key), &value)]);
```

Comment:
If we use references at the end, can we do the same without all those clonings above?